src/blockly/generators/propc/variables.js

```diff
@@ -165,186 +165,131 @@ Blockly.Blocks.array_get = {
-            if (elmnts >= parseInt(initStr, 10) || elmnts < 0) {
-                code = 'WARNING: You are trying to get an element from your array that does not exist!';
+        } else if (this.type === 'array_fill') {
+            elementCount = (this.getFieldValue('NUM').split(',')).length
+        }
+        var arrayName = this.getFieldValue('VAR')
+        var allBlocks = Blockly.getMainWorkspace().getAllBlocks();
+        var initBlock = null;
+        for (var ij = 0; ij < allBlocks.length; ij++) {
+            if (allBlocks[ij].type === 'array_init' && allBlocks[ij].getFieldValue('VAR')) {
+                initBlock = allBlocks[ij];
+                break;
             }
-        } else {
-            code = 'WARNING: The array "' + this.getFieldValue('VAR') + '" has not been initialized!';
+        }
+        if (!initBlock) {
+            code = 'WARNING: The array "' + arrayName + '" has not been initialized!';
+        } else if (elementCount && (elementCount >= parseInt(initBlock.getFieldValue('NUM'), 10) || elementCount < 0)) {
+            code = 'WARNING: You are trying to get an element from your array that does not exist!';
         }
         this.setWarningText(code);
     }
 };
 
-
 /**
  *
  * @type {{init: Blockly.Blocks.array_init.init, helpUrl: *, onchange: Blockly.Blocks.array_init.onchange, sendArrayVal: Blockly.Blocks.array_init.sendArrayVal}}
  */
 Blockly.Blocks.array_init = {
     helpUrl: Blockly.MSG_ARRAYS_HELPURL,
-    // Initialize an array variable
     init: function () {
-        // build a block that has this format:
-        // 'array initialize' {variable_name} 'with' {number} 'elements'
         this.setTooltip(Blockly.MSG_ARRAY_INIT_TOOLTIP);
         this.setColour(colorPalette.getColor('variables'));
         this.appendDummyInput()
-                .appendField('array initialize') // block text
-                .appendField(                           // create a new FieldTextInput object
-                    new Blockly.FieldTextInput(
-                        'list',             // Default variable name
-                        function (a) {      // field content validation
-                            a = a.replace(/ /g, '_').replace(/[^a-zA-Z0-9_]/g, '');
-
-                            // Set the new variable name (in a) for the array
-                            this.sourceBlock_.sendArrayVal(
-                                this.sourceBlock_.getFieldValue('VAR'),
-                                a);
-
-                            return a;
-                        }),
-                    'VAR')
-
-                .appendField("with")    // block text
-                .appendField(
-                    new Blockly.FieldNumber('10', null, null, 1),
-                    'NUM')
-
+                .appendField('array initialize')
+                .appendField(new Blockly.FieldTextInput('list', function (a) {
+                    return a.replace(/ /g, '_').replace(/[^a-zA-Z0-9_]/g, '');
```

Comment:
Same issue with the regex. Can you add a comment to note what the regex is doing?

Reply:
Ditto.